File: backend/src/laboratorios/types/validation.types.ts
export type TemplateType = 'RRA' | 'RRA_FAR';

export interface FieldError {
  column: string;
  field: string;
  message: string;
}

// Tipo interno que usan los validadores
export interface ValidatedRow {
  rowNumber: number;
  isValid: boolean;
  data: Record<string, unknown>;
  errors: FieldError[];
}

// Tipo que devuelve la API al frontend
export interface RowResult {
  rowNumber: number;
  isValid: boolean;
  display: {
    codigoMuestra: string;
    nInforme: string;
    analisis: string;
    valorObtenido: string;
    fechaMuestreo: string;
  };
  errors: FieldError[];
}

export interface ValidationResult {
  templateType: TemplateType;
  totalRows: number;
  validRows: number;
  errorRows: number;
  rows: RowResult[];
}

File: backend-java/src/main/java/cl/sernapesca/laboratorio/validators/ValidationTypes.java
package cl.sernapesca.laboratorio.validators;

import java.util.List;

public class ValidationTypes {

    public record FieldError(
        String column,
        String field,
        String message
    ) {}

    public record RowDisplay(
        String codigoMuestra,
        String nInforme,
        String analisis,
        String valorObtenido,
        String fechaMuestreo
    ) {}

    public record ValidatedRow(
        int rowNumber,
        boolean isValid,
        RowDisplay display,
        List<FieldError> errors
    ) {}

    public record ValidationResult(
        String templateType,
        int totalRows,
        int validRows,
        int errorRows,
        List<ValidatedRow> rows
    ) {}
}